cns/restserver/util.go

```diff
@@ -853,7 +853,10 @@ func (service *HTTPRestService) isNCWaitingForUpdate(
 			"Skipping GetNCVersionStatus check from NMAgent", ncVersion, ncid)
 		return true, types.NetworkContainerVfpProgramPending, ""
 	}
-	nmaProgrammedNCVersionStr, ok := ncVersionList[ncid]
+	// accept both upper and lower GUID from ncid(Swift_ncGUID)
+	// Split ncid by 'Swift_' and lower-case ncGUID(i.e, 89063DBF-AA31) and check if each ncid(Swift_ncGUID, i.e, Swift_89063dbf-aa31) is in ncVersionList
+	ncGUID := strings.ToLower(strings.Split(ncid, cns.SwiftPrefix)[1])
+	nmaProgrammedNCVersionStr, ok := ncVersionList[cns.SwiftPrefix+ncGUID]
```

Comment:
Hi Tim, it was designed by this format "Swift_NCID", and "Swift" is a const value. We probably do not make this change because this is being used. So we just want to partition NCID string here. 
I will use CutPrefix, thanks good learning this lib!

Reply:
Right, I understand that. Without disruption to any existing uses, I believe another type can be introduced that acts as a namespaced map or something. Then that type can be used with the "Swift" const value as-is. The point of this is that this handler function has all of this knowledge about how this map works that needs to be (correctly) replicated anywhere else it appears. In this PR, we've added more special knowledge about this map. Anytime anyone uses this going forward, we need to rely on code inspection to ensure that `cns.SwiftPrefix` is there and that lowercasing is done, and I guarantee you we will miss it someday and introduce a defect. We have an opportunity to make this safer and easier to use, and I think it's worthwhile.

Reply:
I agree, this impl is extremely brittle and we're kicking the can down the road instead of fixing it properly here 👎